File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Events/OnSpawnMonsterMessage.ts
import {IMessage} from "Core/ObserverMessageKit";
import {AbstractMonster} from "mods/ModTheSpire/Scripts/Unit/Monster/AbstractMonster";

export class OnSpawnMonsterMessage implements IMessage{
    public static readonly Id = "OnSpawnMonster";
    public constructor(public m : AbstractMonster) {
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Event/Ids/OnSpawnMonsterMessage.cs
using ModTheSpire.Unit;

namespace ModTheSpire.Event.Ids
{
    ///当生成怪物时
    public class OnSpawnMonsterMessage : IEventHandle
    {
        public AbstractCreature Monster;

        public OnSpawnMonsterMessage(AbstractCreature monster)
        {
            Monster = monster;
        }
    }
}